scripts/build_pkg.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+type=${1}
+build_sub_dir=${2}
+tsconfig='tsconfig.build.json'
+build_dir='build'
+
+shift
+
+if [ -n "${type}" ]; then
+  tsconfig="tsconfig.build.${type}.json"
+
+  if [ -n "${build_sub_dir}" ]; then
+    shift
+    build_dir="${build_dir}/${build_sub_dir}"
+  else
+    build_dir="${build_dir}/${type}"
+  fi
+fi
+
+yarn g:rimraf "${build_dir}"
+yarn g:tsc -p "${tsconfig}" --outDir "${build_dir}" "$@"
+yarn g:tsc-alias -p "${tsconfig}" --outDir "${build_dir}"
+
+if [ -n "${type}" ]; then
+  touch "${build_dir}/package.json"
+fi
```

Comment:
are there any other values for `type` than `esm | cjs`? if not, would the below if statement not take care of creating the package.json? is there a situation we want to create an empty package.json (assuming empty because the below are full replace and not append redirects)?

Reply:
going to revisit this script once all the repos are updated
we may not end up using it since we might just build everything for ESM since we've moved to node 20